survey-sdk/src/main/java/co/nimblehq/survey/sdk/network/MoshiBuilderProvider.kt

```diff
@@ -27,4 +28,6 @@ object MoshiBuilderProvider {
     private fun provideMoshi(): Moshi = moshiBuilder.add((provideJsonApiFactory())).build()
 
     fun getJsonApiConverterFactory() : JsonApiConverterFactory = JsonApiConverterFactory.create(provideMoshi())
+
+    fun getConverterFactory(): MoshiConverterFactory = MoshiConverterFactory.create(moshiBuilder.build())
```

Comment:
A little bit confused here. 

`JsonApiConverterFactory.create(provideMoshi())`
`MoshiConverterFactory.create(moshiBuilder.build())`

Would it be possible to have the `provide` function for `MoshiConverterFactory` instead of a global variable? And I'd suggest passing as the parameter for the decoupling.

Reply:
@hoangnguyen92dn 
`JsonApiConverterFactory.create(provideMoshi())` : Create converter for JsonAPI
`MoshiConverterFactory.create(moshiBuilder.build())` : Create converter for nomarl json.
Perhaps, the function `provideMoshi` is not clear enough, it should be `provideMoshiJsonApi`

Reply:
@doannimble Following the dependency pattern, we should avoid using global variable/methods inside Factory methods, there should be defined dependencies for Factory methods, and it will be easier to do testing later on.

Reply:
@hoangnguyen92dn I've updated, using Factory pattern. 
You're right, in the real project, we need to setup to follow up dependency pattern.
But, as this is the `sample-app` , I'm just trying to make it simple 🙏